src/test/ui/consts/const-mut-refs/mut_ref_in_final_ok.rs

```diff
@@ -0,0 +1,22 @@
+#![feature(const_mut_refs)]
+#![feature(const_fn)]
+#![feature(raw_ref_op)]
+
+use std::cell::UnsafeCell;
+struct NotAMutex<T>(UnsafeCell<T>);
+
+unsafe impl<T> Sync for NotAMutex<T> {}
+
+const FOO: NotAMutex<&mut i32> = NotAMutex(UnsafeCell::new(&mut 42));
+//~^ ERROR temporary value dropped while borrowed
+
+// `BAR` works, because `&42` promotes immediately instead of relying on
+// "final value lifetime extension".
```

Comment:
I think we call it the "enclosing scope rule" in our docs.

Reply:
This comment was meant to explain why the situation is different from the other situations in the same file. Thus I brought it up again.

Reply:
Okay, I see. But please use consistent terminology.